fix： 服务不在服务发现中，判断IsMaster panic

问题描述
===========
<!-- 这里写问题描述 -->

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8c64f1]

goroutine 46 [running]:
sync.(*RWMutex).RLock(...)
	/home/golang/src/sync/rwmutex.go:48
configcenter/src/apimachinery/discovery.(*server).IsMaster(0x0, 0xc000b27de0, 0x15, 0x0)
	/home/gosrc/src/configcenter/src/apimachinery/discovery/server.go:73 +0x31
configcenter/src/apimachinery/discovery.(*discover).IsMaster(0xc0005b4008, 0xed45b00c9)
	/home/gosrc/src/configcenter/src/apimachinery/discovery/discovery.go:141 +0xac
configcenter/src/scene_server/synchronize_server/logics.(*Logics).TriggerSynchronize(0xc0000f8100, 0xeceba0, 0xc0005c2340, 0xc00022ca20)
	/home/gosrc/src/configcenter/src/scene_server/synchronize_server/logics/synchronize.go:64 +0x294
created by configcenter/src/scene_server/synchronize_server/service.(*Service).InitBackground



重现方法
================
<!-- 列出如何重现的方法或操作步骤 -->


<!-- **重要提醒**:  请优先尝试最新发布的版本 (发布清单： https://github.com/Tencent/bk-cmdb/releases), 如果问题不能在最新发布的版本里重现，说明此问题已经被修复。 -->


关键信息
=========

<!--  **重要提醒**: 这些关键信息会辅助我们快速定位问题。 -->

请提供以下信息:

 - [x] bk-cmdb   版本: 
 <!-- 示例： ./cmdb/cmdb-apiserver --version 输出的内容:
Version  : oc-19.01.30
Tag      : 
Branch   : v3.3.x
BuildTime: 2019-01-30T11:29:27+0800
GitHash  : a99aabedf7a369387423ad0ab013730c88f4a7b6
RunMode  : product
 -->
 - [ ] Redis     版本: <!-- `示例： 3.2.11`  -->
 - [ ] MongoDB   版本: <!-- `示例： 2.8.0` -->
 - [ ] ZooKeeper 版本: <!-- `示例： 3.4.11` -->
 - [ ] 操作系统      : <!-- `示例： Centos 5 (x64>` -->
 - [ ] bk-cmdb 异常日志

--- a/src/apimachinery/discovery/server.go
+++ b/src/apimachinery/discovery/server.go
@@ -51,6 +51,9 @@ type server struct {
 }
 
 func (s *server) GetServers() ([]string, error) {
+	if s == nil {
+		return []string{}, nil
+	}
 	s.RLock()
 	defer s.RUnlock()
 
@@ -70,6 +73,9 @@ func (s *server) GetServers() ([]string, error) {
 
 // IsMaster 判断当前进程是否为master 进程， 服务注册节点的第一个节点
 func (s *server) IsMaster(strAddrs string) bool {
+	if s == nil {
+		return false
+	}
 	s.RLock()
 	defer s.RUnlock()
 	if 0 < len(s.servers) {

--- a/src/common/types/serverInfo.go
+++ b/src/common/types/serverInfo.go
@@ -148,5 +148,8 @@ type EventServInfo struct {
 
 // Address convert struct to host address
 func (s *ServerInfo) Address() string {
+	if s == nil {
+		return ""
+	}
 	return fmt.Sprintf("%s://%s:%d", s.Scheme, s.IP, s.Port)
 }

